### ISSUE TITLE
Temperature in slide-level visual-caption pretraining

### ISSUE THREAD

**Issue by lanfz2000** [mahmoodlab/TITAN#10]
The code in TITAN is:
    def zero_shot(self, slide_embedding: torch.Tensor, classifier: torch.Tensor) -> torch.Tensor:
        '''
        zero-shot inference for whole-slide image
        Args:
            slide_embedding: torch.Tensor, shape (1, D)
            classifier: torch.Tensor, shape (D, num_classes)
        '''
        device = slide_embedding.device

        slide_embedding = slide_embedding.float().to(device).unsqueeze(0)
        slide_embedding = slide_embedding @ self.vision_encoder.proj
        slide_embedding = F.normalize(slide_embedding, dim=-1)

        logits = slide_embedding @ classifier
        probs = F.softmax(logits, dim=-1)
        return probs

I wonder where is the temperature scale, as the probability of the 46-class is quite close.

**Comment by andrewsong90** [comment]
Hi @lanfz2000 

The code snippet you are mentioning is for inference (We haven't released the pretraining code for vision-language).
As you rightly said, the temperature parameter is an important consideration for the pretraining stage. For the inference stage however, since we are using balanced accuracy and the softmax operation **does not change** the order of predictions, default temperature should be good to use.

Hope this helps!
